source/server-selection/server-selection-tests.rst

```diff
@@ -256,3 +256,11 @@ return a set of three suitable servers within the latency window::
 
 Drivers SHOULD check that their implementation selects one of ``primary``, ``secondary_1``,
 and ``secondary_2`` at random.
+
+Application-Provided Server Selector
```

Comment:
If anyone has an idea for a more specific test, let me know.  I was thinking about something like:

Run against a replica set with more than one secondary.  Create a custom server selector that selects the one with the highest port number.  Execute 100 queries with secondary read preference and assert with command monitoring that all of them execute on the secondary with highest port number.

Reply:
Makes sense to me.

Reply:
Done

Reply:
Done